tokio/build.rs

```diff
@@ -1,192 +0,0 @@
-                // compiles.
-                //
-                // The oldest nightly that supports the feature is 2022-06-20.
-                if ac.probe_expression(CONST_MUTEX_NEW_PROBE) {
-                    enable_const_mutex_new = true;
-                }
-            }
-
-            // The `AsFd` family of traits were made stable in 1.63.
-            if ac.probe_rustc_version(1, 64) {
-                enable_as_fd = true;
-            } else if ac.probe_rustc_version(1, 63) {
-                // This compiler claims to be 1.63, but there are some nightly
-                // compilers that claim to be 1.63 without supporting the
-                // feature. Explicitly probe to check if code using them
-                // compiles.
-                //
-                // The oldest nightly that supports the feature is 2022-06-16.
-                if ac.probe_expression(AS_FD_PROBE) {
-                    enable_as_fd = true;
-                }
-            }
-        }
-
-        Err(e) => {
-            // If we couldn't detect the compiler version and features, just
-            // print a warning. This isn't a fatal error: we can still build
-            // Tokio, we just can't enable cfgs automatically.
-            println!(
-                "cargo:warning=tokio: failed to detect compiler features: {}",
-                e
-            );
-        }
-    }
-
-    if !enable_const_thread_local {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_const_thread_local"
-        autocfg::emit("tokio_no_const_thread_local")
-    }
-
-    if !enable_target_has_atomic {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_target_has_atomic"
-        autocfg::emit("tokio_no_target_has_atomic")
-    }
-
-    if !enable_const_mutex_new {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_const_mutex_new"
-        autocfg::emit("tokio_no_const_mutex_new")
-    }
-
-    if !enable_as_fd {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_as_fd"
-        autocfg::emit("tokio_no_as_fd");
-    }
-
-    if target_needs_atomic_u64_fallback {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_atomic_u64"
-        autocfg::emit("tokio_no_atomic_u64")
-    }
```

Comment:
Sure, I will do this in a separate PR.

Though `tokio_no_atomic_u64` and `tokio_no_const_mutex_new` is still used in CI for `no-atomic-u64-{check, test}` and `cross-test-with-parking_lot`.

Reply:
> Though `tokio_no_atomic_u64` and `tokio_no_const_mutex_new` is still used in CI for `no-atomic-u64-{check, test}` and `cross-test-with-parking_lot`.

CI jobs using `tokio_no_const_mutex_new` are for testing cases that occur only in old rustc, so it is safe to remove it.
`tokio_no_atomic_u64` is also used for portability testing, so removing it may requires slightly adjusting the CI config.

Reply:
Thanks, I will open a PR to remove everything but `tokio_no_atomic_u64` then, since I am still not sure how to remove `tokio_no_atomic_u64` cfg (could definitely rename it though).